astrocut/CubeCutout.py

```diff
@@ -0,0 +1,564 @@
+
+        Parameters
+        ----------
+        table_header : `~astropy.io.fits.Header`
+            The table header for the cutout table that will be modified in place to include 
+            WCS information.
+        wcs_dict : dict
+            Dictionary of wcs keyword/value pairs to be added to each array column in the 
+            cutout table header.
+        """
+        # Mapping of FITS table keywords to descriptions
+        keyword_comments = {
+            'TTYPE': 'column name',
+            'TFORM': 'column format',
+            'TUNIT': 'unit',
+            'TDISP': 'display format',
+            'TDIM': 'multi-dimensional array spec',
+            'TNULL': 'null value'
+        }
+
+        for kwd in table_header:
+            # Set header comment if the keyword matches
+            for key, comment in keyword_comments.items():
+                if key in kwd:
+                    table_header.comments[kwd] = comment
+                    break
+            
+            # If the column is a 2D array (indicated by 'TDIM'), add WCS info
+            if 'TDIM' in kwd and kwd[:-1] == 'TTYPE':
+                for wcs_key, (val, com) in wcs_dict.items():
+                    table_header.insert(kwd, (wcs_key.format(int(kwd[-1]) - 1), val, com))
+
+    def _add_img_kwds(self, table_header: fits.Header):
+        """
+        Adding extra keywords to the table header.
+
+        Parameters
+        ----------
+        table_header : `~astropy.io.fits.Header`
+            The table header to add keywords to.  It will be modified in place.
+        """
+        # Add image keywords to the table header
+        for key, value in self._img_kwds.items():
+            if key not in self._skip_kwds:
+                table_header[key] = tuple(value)
+
+    def _apply_header_inherit(self, hdu_list: fits.HDUList):
+        """
+        The INHERIT keyword indicated that keywords from the primary header should be duplicated in 
+        the headers of all subsequent extensions.  This function performs this addition in place to 
+        the given HDUList.
+        
+        Parameters
+        ----------
+        hdu_list : `~astropy.io.fits.HDUList`
+            The hdu list to apply the INHERIT keyword to.
+        """
+        primary_header = hdu_list[0].header
+        reserved_kwds = {'COMMENT', 'SIMPLE', 'BITPIX', 'EXTEND', 'NEXTEND'}
+
+        for hdu in hdu_list[1:]:
+            header = hdu.header
+            if header.get('INHERIT', False):
+                for kwd, val in primary_header.items():
+                    if (kwd not in header) and (kwd not in reserved_kwds):
+                        header[kwd] = (val, primary_header.comments[kwd])
+
+    @abstractmethod
+    def _build_tpf(self, cube_fits: fits.HDUList, cutout: 'CubeCutoutInstance') -> fits.HDUList:
+        """
+        Build the cutout target pixel file (TPF).
+
+        This method is abstract and should be defined in subclasses.
+        """
+        pass
```

Comment:
Will `CubeCutout` be used as the parent for any non-TESS cube cutout classes, e.g. JWST or MaNGA cubes?  If so,  maybe this should be moved to `TessCubeCutout`, or moved into some intermediate `TessBase` class if it needs to be an `abstractmethod`.